inc/Engine/CriticalPath/AsyncCSS.php

```diff
@@ -0,0 +1,376 @@
+<?php
+
+namespace WP_Rocket\Engine\CriticalPath;
+
+use DOMElement;
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Engine\DOM\HTMLDocument;
+use WP_Rocket\Engine\DOM\Attribute;
+
+class AsyncCSS {
+
+	/**
+	 * Instance of Critical CSS.
+	 *
+	 * @var Critical_CSS
+	 */
+	private $critical_css;
+
+	/**
+	 * Instance of options.
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Instance of the DOM.
+	 *
+	 * @var HTMLDocument
+	 */
+	private $dom;
+
+	/**
+	 * <noscript> element.
+	 *
+	 * @var DOMElement
+	 */
+	private $noscript;
+
+	/**
+	 * Array of URLs to exclude.
+	 *
+	 * @var array
+	 */
+	private $css_urls_to_exclude;
+
+	/**
+	 * The "onload" attribute defaults.
+	 *
+	 * @var array
+	 */
+	private $onload_defaults = [
```

Comment:
This does not "feel" right. Shouldn't it be something like `attribute_defaults` to make it feel way global than `$onload_defaults` ? 🤔 Because it contains also `this.rel` which is not related to `onload` attributes or changes required.

Reply:
This is only used for the `onload` attribute values. It's used to create this:

```html
onload="this.onload=null;this.media='all';this.rel='stylesheet'"
```

Notice that the `rel` in there too, which will change from `rel="preload"` to `rel="stylesheet"` when "onload" event fires.